tck/methods/key.ts

```diff
@@ -0,0 +1,207 @@
+                const pk = PrivateKey.generateED25519();
+                privateKey = pk.toStringDer();
+                publicKey = pk.publicKey.toStringDer();
+            } else {
+                const pk = PrivateKey.generateECDSA();
+                privateKey = pk.toStringDer();
+                publicKey = pk.publicKey.toStringDer();
+            }
+
+            if (isList) {
+                response.privateKeys.push(privateKey);
+            }
+
+            return { key: publicKey, privateKeys: response.privateKeys };
+
+        case AccountKey.KEY_LIST:
+        case AccountKey.THRESHOLD_KEY:
+            const keyList = new KeyList();
+
+            for (const keyParam of params.keys!) {
+                const { key: keyStr } = processKeyRecursively(
+                    keyParam,
+                    response,
+                    true,
+                );
+                const key = getKeyFromString(keyStr);
+                if (key) {
+                    keyList.push(key);
+                }
+            }
+            if (params.type === AccountKey.THRESHOLD_KEY) {
+                keyList.setThreshold(params.threshold!);
+            }
+
+            return {
+                key: convertToKeyListHex(keyList) || "",
+                privateKeys: response.privateKeys,
+            };
+
+        case AccountKey.EVM_ADDRESS:
+            if (params.fromKey) {
+                try {
+                    const key = getKeyFromString(params.fromKey);
+                    if (key instanceof PublicKey) {
+                        return {
+                            key: key.toEvmAddress(),
+                            privateKeys: response.privateKeys,
+                        };
+                    } else if (key instanceof PrivateKey) {
+                        return {
+                            key: key.publicKey.toEvmAddress(),
+                            privateKeys: response.privateKeys,
+                        };
+                    } else {
+                        return invalidParamError<KeyGenerationResponse>(
+                            "fromKey for evmAddress is not ECDSAsecp256k1",
+                        );
+                    }
+                } catch (error) {
+                    return invalidParamError<KeyGenerationResponse>(
+                        error.message,
+                    );
+                }
+            }
+            try {
+                const privateKey = PrivateKey.generateECDSA();
+                return {
+                    key: privateKey.publicKey.toEvmAddress(),
+                    privateKeys: [privateKey.toStringDer()],
+                };
+            } catch (error) {
+                return invalidParamError<KeyGenerationResponse>(error.message);
+            }
+
+        default:
+            return invalidParamError<KeyGenerationResponse>(
+                "key type not recognized",
+            );
+    }
+};
```

Comment:
If there's one unpopular opinion that I am willing to die on the hill for - it's that switch cases are terrible! 

Lookup table are like a million times better. If you haven't used one I can help you. Best case is that you should be using polymorphism but in your case that's not possible.